### ISSUE TITLE
ID from m3u8 list not matching EPG id

### ISSUE THREAD

**Issue by alanmilinovic** [kodi-pvr/pvr.iptvsimple#692]
Hi, why is IPTV Simple Client not case sensitive when matching tvg-id with channel id from EPG?

Example m3u8:
```
#EXTINF:-1 tvg-id="rtl" ...
#EXTINF:-1 tvg-id="RTL" ...
```

Example xmltv:
```
<channel id="rtl"><display-name> ...
<channel id="rtl"><display-name> ...
<channel id="RTL"><display-name> ...
```

So both channels called "rtl" and "RTL" will have `<channel id="RTL">` from EPG.
Can you please make it case sensitive?

**Comment by phunkyfish** [comment]
The add-on has always worked this way. The problem with changing it now means that behaviour would change for everyone.

**Comment by alanmilinovic** [comment]
The question is more what is the correct way. ID means uniqueidentifier and it should be, if you ask me case sensitive because of that.

**Comment by alanmilinovic** [comment]
Maybe adding an optional function in addon settings where people can chose and enable it if they want?

**Comment by phunkyfish** [comment]
If the add-on has worked this way always the support cost of changing the behaviour must be considered. It will be me who has to support users when they don’t know why their EPG is not working as expected.

That being said, it should be easy enough to provide a setting to allow case sensitivity to be switched on/off.

**Comment by alanmilinovic** [comment]
Great, glad we agreed on it. Thank you.

**Comment by alanmilinovic** [comment]
Do you have time to code it? Need any help with testing?

**Comment by phunkyfish** [comment]
I can probably get around to it this week. The change itself is easy, most of the work is adding the setting.

Will post a testbuild here when ready. What platform can you test on?

**Comment by alanmilinovic** [comment]
I can test it on debian 11 (amd64 x86_64).

**Comment by phunkyfish** [comment]
Can you build the addon yourself? We don’t provide testbuilds for anything other than Ubuntu.

If not maybe @basilgello can help.

**Comment by phunkyfish** [comment]
PR: https://github.com/kodi-pvr/pvr.iptvsimple/pull/693

All the non linux builds are here: https://jenkins.kodi.tv/blue/organizations/jenkins/kodi-pvr%2Fpvr.iptvsimple/detail/PR-693/2/artifacts

**Comment by alanmilinovic** [comment]
I can create ubuntu virtual machine and test it there no problem.

**Comment by alanmilinovic** [comment]
Just tested on Windows and it is working great. Thank you very much!

For now I will use it on my Windows machine and once Kodi version 20 become available for linux debian (curently it is in deb-multimedia test repo), will use it there as well. At the moment I am still on Kodi Matrix on my Linux machine.

**Comment by alanmilinovic** [comment]
> Can you build the addon yourself? We don’t provide testbuilds for anything other than Ubuntu.
> 
> If not maybe @basilgello can help.

If I build it myself, will it work on Kodi Matrix? I guess not, if that is the case then I need to wait for Kodi Nexus debian package to become stable.

**Comment by phunkyfish** [comment]
No, as it’s a binary addon it must be built based on the APIs for that version of kodi.

But I can create a matrix version easily enough if you can test it.

**Comment by phunkyfish** [comment]
Matrix PR: https://github.com/kodi-pvr/pvr.iptvsimple/pull/694
Matrix Testbuilds: https://jenkins.kodi.tv/blue/organizations/jenkins/kodi-pvr%2Fpvr.iptvsimple/detail/PR-694/2/artifacts

**Comment by alanmilinovic** [comment]
Looks like build was not successful, can you trigger build again please? In the meantime I will do it on debian 11.

**Comment by basilgello** [comment]
I can upload v20 into bullseye-backports but I am waiting for completion of the keyring change. It should have been done already but something stalls.

**Comment by alanmilinovic** [comment]
And where can I find binary files after `make` command?

**Comment by alanmilinovic** [comment]
I think I found it, but I am getting error.
`The dependency on inputstream.rtmp version 0.0.0 could not be satisfied.`

I know how to add rtmp package but not sure why version 0.0.0., any idea?

**Comment by basilgello** [comment]
Just install kodi-inputstream-adaptive and inputstream-rtmp package (you are still on Matrix, so that'd be sufficient)

**Comment by alanmilinovic** [comment]
I had to restart Kodi, so that the system recognize the rtmp addon. It is all working nice, thank you both very much.
